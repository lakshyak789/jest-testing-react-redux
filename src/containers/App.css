.container {
  text-align: center;
}
.amount {
  font-size: 20px;
  margin-right: 15px;
}
.input-box {
  border: none;
  outline: none;
  width: 240px;
  height: 30px;
  font-size: 20px;
  padding-left: 10px;
  background-color: rgb(223, 221, 221);
}
.btn-wrap {
  margin-top: 35px;
  margin-left: 50px;
}
.btn-wrap .deposit-balance {
  margin-left: 10px;
  padding: 10px 20px;
  background-color: rgb(97, 243, 97);
  outline: none;
  border: none;
  cursor: pointer;
  box-shadow: 1px cadetblue;
}
.btn-wrap .withdraw-balance {
  margin-left: 10px;
  padding: 10px 20px;
  background-color: rgb(243, 175, 97);
  outline: none;
  border: none;
  cursor: pointer;
  box-shadow: 1px cadetblue;
}
