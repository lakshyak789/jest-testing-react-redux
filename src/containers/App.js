import React, { Component } from "react";
import "./App.css";
import { connect } from "react-redux";
import { deposit, withdraw } from "../actions/balance";
export class App extends Component {
  state = { balance: 0 };
  handeChange(e) {
    if (isNaN(parseInt(e.target.value))) {
      if (e.target.value === "") {
        this.setState({ balance: 0 });
      }
    } else {
      this.setState({ balance: parseInt(e.target.value) });
    }
  }
  deposit() {
    this.props.deposit(this.state.balance);
    this.setState({ balance: 0 });
  }
  withdraw() {
    this.props.withdraw(this.state.balance);
    this.setState({ balance: 0 });
  }
  render() {
    return (
      <div className="container">
        <h1 className="balance">Balance is: {this.props.balance}</h1>
        <label className="amount">Enter Amount:</label>
        <input
          type="text"
          className="input-box"
          onChange={this.handeChange.bind(this)}
          value={this.state.balance}
        />
        <div className="btn-wrap">
          <button className="deposit-balance" onClick={this.deposit.bind(this)}>
            Deposit
          </button>
          <button
            className="withdraw-balance"
            onClick={this.withdraw.bind(this)}
          >
            Withdraw
          </button>
        </div>
      </div>
    );
  }
}

export default connect(
  state => {
    return { balance: state };
  },
  { deposit, withdraw }
)(App);
