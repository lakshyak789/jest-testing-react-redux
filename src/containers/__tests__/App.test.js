import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { App } from "../App";
configure({ adapter: new Adapter() });
describe("App component", () => {
  const mockdeposit = jest.fn();
  const mockwithdraw = jest.fn();
  const props = { balance: 20, deposit: mockdeposit, withdraw: mockwithdraw };
  const app = shallow(<App {...props} />);
  it("should render properly", () => {
    expect(app).toMatchSnapshot();
  });
  it("should display balance from props", () => {
    expect(app.find(".balance").text()).toEqual("Balance is: 20");
  });
  describe("on change of textbox value when empty or with Nan", () => {
    beforeEach(() => {
      app.find(".input-box").simulate("change", { target: { value: "" } });
    });
    it("should update the state", () => {
      expect(app.state().balance).toEqual(0);
    });
  });
  describe("on change of textbox value", () => {
    const updatedbalance = 20;
    beforeEach(() => {
      app
        .find(".input-box")
        .simulate("change", { target: { value: updatedbalance } });
    });
    it("should update the state", () => {
      expect(app.state().balance).toEqual(parseInt(updatedbalance));
    });
    describe("on click of deposit button", () => {
      beforeEach(() => {
        app.find(".deposit-balance").simulate("click");
      });
      it("should call the deposit action ", () => {
        expect(mockdeposit).toHaveBeenCalledWith(parseInt(updatedbalance));
      });
    });
    describe("on click of withdraw button", () => {
      beforeEach(() => {
        app.find(".withdraw-balance").simulate("click");
      });
      it("should call the withdraw button", () => {
        expect(mockwithdraw).toHaveBeenCalledWith(parseInt(updatedbalance));
      });
    });
  });
});
