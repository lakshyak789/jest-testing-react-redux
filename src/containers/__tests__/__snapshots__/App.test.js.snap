// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    balance={20}
    deposit={[MockFunction]}
    withdraw={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="balance"
        >
          Balance is: 
          20
        </h1>,
        <label
          className="amount"
        >
          Enter Amount:
        </label>,
        <input
          className="input-box"
          onChange={[Function]}
          type="text"
          value={0}
        />,
        <div
          className="btn-wrap"
        >
          <button
            className="deposit-balance"
            onClick={[Function]}
          >
            Deposit
          </button>
          <button
            className="withdraw-balance"
            onClick={[Function]}
          >
            Withdraw
          </button>
        </div>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Balance is: ",
            20,
          ],
          "className": "balance",
        },
        "ref": null,
        "rendered": Array [
          "Balance is: ",
          20,
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Enter Amount:",
          "className": "amount",
        },
        "ref": null,
        "rendered": "Enter Amount:",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "input-box",
          "onChange": [Function],
          "type": "text",
          "value": 0,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="deposit-balance"
              onClick={[Function]}
            >
              Deposit
            </button>,
            <button
              className="withdraw-balance"
              onClick={[Function]}
            >
              Withdraw
            </button>,
          ],
          "className": "btn-wrap",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Deposit",
              "className": "deposit-balance",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Deposit",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Withdraw",
              "className": "withdraw-balance",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Withdraw",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="balance"
          >
            Balance is: 
            20
          </h1>,
          <label
            className="amount"
          >
            Enter Amount:
          </label>,
          <input
            className="input-box"
            onChange={[Function]}
            type="text"
            value={0}
          />,
          <div
            className="btn-wrap"
          >
            <button
              className="deposit-balance"
              onClick={[Function]}
            >
              Deposit
            </button>
            <button
              className="withdraw-balance"
              onClick={[Function]}
            >
              Withdraw
            </button>
          </div>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Balance is: ",
              20,
            ],
            "className": "balance",
          },
          "ref": null,
          "rendered": Array [
            "Balance is: ",
            20,
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Enter Amount:",
            "className": "amount",
          },
          "ref": null,
          "rendered": "Enter Amount:",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input-box",
            "onChange": [Function],
            "type": "text",
            "value": 0,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="deposit-balance"
                onClick={[Function]}
              >
                Deposit
              </button>,
              <button
                className="withdraw-balance"
                onClick={[Function]}
              >
                Withdraw
              </button>,
            ],
            "className": "btn-wrap",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Deposit",
                "className": "deposit-balance",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Deposit",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Withdraw",
                "className": "withdraw-balance",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Withdraw",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
